.github/workflows/release.yml

```diff
@@ -0,0 +1,39 @@
+name: CDK Deployment
+
+on:
+  push:
+    branches:
+      - main
+      - stg
+      - dev
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        repository: 'AI4Organization/faiss-backend-with-langchain-python-coreservices'
+        ref: ${{ github.ref }}
+        ssh-key: ${{ secrets.SSH_KEY }}
+
+    - name: Set up AWS CLI
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: ${{ secrets.AWS_REGION }}
+
+    - name: Install AWS CDK CLI
+      run: npm install -g aws-cdk
```

Comment:
**Suggestion:** Consider using 'actions/setup-node' to install Node.js before running 'npm install' to ensure the correct Node.js version is used.
```suggestion
    - name: Setup Node.js
      uses: actions/setup-node@v2
      with:
        node-version: '14' # Specify the Node.js version to use
    - name: Install AWS CDK CLI
      run: npm install -g aws-cdk
```